frontend/src/components/MessagesPage.vue

```diff
@@ -1,36 +1,33 @@
 <template>
   <v-container id="messages" fluid class="pa-0 ma-0" height="100%">
-    <v-row v-if="!allMessages" class="ma-0">
+    <v-row v-if="!allMessages || allMessages.length === 0" class="ma-0">
       <Spinner style="width: 100%" />
     </v-row>
     <v-row v-else fluid class="mx-4">
       <v-row>
         <v-col id="messages-summary" fluid class="pa-0" :cols="4">
           <v-card tile style="border-right: 1px solid lightgrey" :height="fitScreenHeight()" class="pa-0 elevation-0">
             <v-data-table
-              :headers="headers"
+              :headers="columns"
               :items="messages"
-              mobile-breakpoint="960"
-              fluid
+              :mobile-breakpoint="960"
               :height="fitScreenHeight()"
               fixed-header
-              :row-props="getMessageStyle"
-              disable-pagination
+              :item-value="item => item.messageId"
+              :items-per-page="-1"
               hide-default-footer
-              item-key="messageId"
-              single-select
               @click:row="rowClickHandler"
             >
-              <template #item.isRead="{ item }">
-                <p v-if="item.isRead">Read</p>
-                <p v-else>Unread</p>
-              </template>
+            <template v-slot:item.isRead="{ item }">
+              <p v-if="item.isRead">Read</p>
+              <p v-else>Unread</p>
+            </template>
             </v-data-table>
           </v-card>
         </v-col>
         <v-col v-if="message.sender" id="messages-content" fluid class="pa-0" :cols="8">
           <v-card class="pa-4 overflow-auto elevation-0" fluid tile :height="fitScreenHeight()">
-            <v-card-title class="pa-0">
+            <v-card-title class="pa-0 d-flex">
```

Comment:
The message's subject is currently hidden in message header. I think we should remove d-flex then add 2 v-row: 1 for sender and date Received, 1 for subject. (Sender and Date Received should be in 1 line, and Subject should be in the next line)